# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-am-build gtk2-build"

PKG_CONFIG_PATH=/dfb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.12)
aclocal; autoconf; automake -a -c
PKG_CONFIG_PATH=/dfb/lib/pkgconfig LDFLAGS=-Wl,-rpath,/dfb/lib ./configure --prefix=/dfb

# build
make

# install
install -d $DESTDIR/dfb/bin
install src/gtktetris $DESTDIR/dfb/bin

# build.pc
install -d $DESTDIR/dfb/share/pkgconfig
cat > $DESTDIR/dfb/share/pkgconfig/gtktetris-build.pc << EOF
Name: gtktetris
Version: 20201213
Description: Tetris clone with GTK interface
Requires: $REQUIRES

exec=\\
/dfb/bin/gtktetris
EOF

exit

# patch
--- gtktetris-20201213.orig/configure.ac
+++ gtktetris-20201213/configure.ac
@@ -1,12 +1,9 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_INIT([gtktetris], [0.6.2b])
-AM_CONFIG_HEADER([config.h])
 AM_INIT_AUTOMAKE([-Wall foreign])
-AC_CONFIG_MACRO_DIR([m4])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 m4_ifdef([AM_MAINTAINER_MODE], [AM_MAINTAINER_MODE])
-LT_INIT
 
 AC_ISC_POSIX
 AC_PROG_CC
@@ -23,7 +20,7 @@
 if test "x$enable_gtk3" = "xyes" ; then
   gtk_modules="gtk+-3.0 >= 3.0.0"
 else
-  gtk_modules="gtk+-2.0 >= 2.14.0"
+  gtk_modules="gtk+-2.0 >= 2.10.0"
 fi
 PKG_CHECK_MODULES(GTK, [$gtk_modules])
 AC_SUBST(GTK_CFLAGS)
--- gtktetris-20201213.orig/src/gtkcompat.h
+++ gtktetris-20201213/src/gtkcompat.h
@@ -353,6 +353,8 @@
 #	define GDK_KEY_Up GDK_Up
 #	define GDK_KEY_space GDK_space
 #	define GDK_KEY_Down GDK_Down
+#	define GDK_KEY_Right GDK_Right
+#	define GDK_KEY_Left GDK_Left
 #	define GDK_KEY_Return GDK_Return
 #	define GDK_KEY_exclam GDK_exclam
 #	define GDK_KEY_BackSpace GDK_BackSpace

# source
https://github.com/wader/gtktetris/archive/4d394f8fb76b9a7e2114ac415621b1818c1d5f8b/gtktetris-20201213.tar.gz
